java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/ShareFileService.java

```diff
@@ -65,16 +65,20 @@ public String generateSharedCode(PremiumUserFilesRepository.UserFile userFile, b
 		return uniqueCode;
 	}
 
-	private String generateUniqueCode() {
-		String uniqueCode;
+	private UUID generateUniqueCode() {
+		UUID uniqueCode;
 		do {
-			uniqueCode = UUID.randomUUID().toString();
+			uniqueCode = UUID.randomUUID();
 		} while (shareFileRepository.findByUuid(uniqueCode) != null);
 		return uniqueCode;
 	}
 
 	@Transactional
-	public ShareFileRepository.ShareFile createShareFile(PremiumUserFilesRepository.UserFile userFile, boolean publicAccess, String uniqueCode) {
+	public ShareFileRepository.ShareFile createShareFile(PremiumUserFilesRepository.UserFile userFile, boolean publicAccess, UUID uniqueCode) {
+		ShareFileRepository.ShareFile existingFile = getFileByOwnerAndFilepath(userFile.userid, userFile.name);
+		if (existingFile != null) {
+			return existingFile;
+		}
```

Comment:
1. Please add log (this is unusual case, almost error)
2. Is uniqueCode ignored in this case?